addons/block_code/ui/block_canvas/block_canvas.gd

```diff
@@ -131,37 +131,16 @@ func _drop_obj_property(at_position: Vector2, data: Variant) -> void:
 	# Prepare a Variable block to set / get the property's value according to
 	# the modifier KEY_CTRL pressing.
 	var block_definition: BlockDefinition
-	var property_type = typeof(property_value)
+	var block_name: String
+	var arguments := {"value": property_value}
+
 	if _modifier_ctrl:
-		var type_string: String = Types.VARIANT_TYPE_TO_STRING[property_type]
-		block_definition = (
-			BlockDefinition
-			. new(
-				&"%s_set_%s" % [object_name, property_name],
-				object_name,
-				"Set the %s property" % property_name,
-				"Variables",
-				Types.BlockType.STATEMENT,
-				property_type,
-				"set %s to {value: %s}" % [property_name.capitalize().to_lower(), type_string],
-				"%s = {value}" % property_name,
-				{"value": property_value},
-			)
-		)
+		block_name = &"%s_set_%s" % [object_name, property_name]
+		block_definition = _context.block_script.get_obj_property_block_definition(block_name, property_value)
```

Comment:
I really don't like how we construct a string and then pass it to a function which immediately splits it back up into the 3 pieces we had here.